src/providers/okta.ts

```diff
@@ -0,0 +1,104 @@
+import { OAuth2Client } from "oslo/oauth2";
+import type { OAuth2ProviderWithPKCE } from "../index.js";
+import { TimeSpan, createDate } from "oslo";
+
+export class Okta implements OAuth2ProviderWithPKCE {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		oktaDomain: string,
+		clientId: string,
+		clientSecret: string,
+		redirectURI: string,
+		options: {
+			authServerId?: string;
+		}
+	) {
+		const authorizeEndpoint = `${oktaDomain}/oauth2/${
```

Comment:
Nitpick, but can you do something like this instead?

```ts
let authorizeEndpoint: string;
let tokenEndpoint: string
if (options.authServerId !== undefined) {
  authorizeEndpoint = "";
  tokenEndpoint = "";
} else {
  // ...
}
```

Reply:
Fixed in latest commit